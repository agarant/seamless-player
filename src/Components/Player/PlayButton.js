import React from "react";
import Button from "material-ui/Button";
import PlayIcon from "material-ui-icons/PlayCircleFilled";
import PauseIcon from "material-ui-icons/PauseCircleFilled";
import IconButton from "material-ui/IconButton";
import { withTheme } from "styled-components";

class PlayButton extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      playing: false
    };
  }

  render() {
    return (
      <IconButton
        onClick={this.props.onClick}
        color="primary"
        style={{ transform: "scale(2.3)" }}
        disabled={this.props.disabled}
      >
        {this.props.playing ? <PauseIcon /> : <PlayIcon />}
      </IconButton>
    );
  }
}

export default withTheme(PlayButton);
