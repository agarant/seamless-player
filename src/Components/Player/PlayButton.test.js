import React from "react";
import renderer from "react-test-renderer";
import PlayButton from "./PlayButton";

test("<PlayButton /> renders the proper icon when playing", () => {
  const tree = renderer
    .create(<PlayButton playing={true} disabled={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
