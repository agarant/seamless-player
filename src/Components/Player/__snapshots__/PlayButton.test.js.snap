// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayButton /> renders the proper icon when playing 1`] = `
<button
  className="MuiButtonBase-root-10 MuiIconButton-root-1 MuiIconButton-colorPrimary-4"
  disabled={false}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  style={
    Object {
      "transform": "scale(2.3)",
    }
  }
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label-7"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-12 MuiIconButton-icon-8"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-1 14H9V8h2v8zm4 0h-2V8h2v8z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-13"
  />
</button>
`;
